statsrelay/src/backends.rs

```diff
@@ -60,6 +60,7 @@ impl BackendsInner {
 
     fn remove_backend(&mut self, name: &str) -> anyhow::Result<()> {
         self.backends.remove(name);
+        // With the backend removed should we ensure it's flushed?
```

Comment:
From my understanding, dropping the backend here will flush it given the current architecture. Unlike when we call `drain_backends(...)` though, we don't flush the processors because we aren't completely shutting down, so the processors will continue operating with the rest of the backends.  However, we don't wait for these removed backend tasks to terminate when shutting down. This means that if we call `remove_backend(...)` and then `drain_backends(...)` in relatively quick succession, we could shut down before the removed backend terminates.  Is that the issue you're referencing here?

Reply:
fyi https://github.com/lyft/statsrelay/pull/80

Reply:
I made the comment as we suspect there's an issue with these backends persisting after a ring swap and as I was walking through the code it seemed possible just removing the backend from the hashmap didn't really guarantee a clean shutdown and cleanup of the backend. I'll check out your PR, thanks 👍